esp-hal/src/lib.rs

```diff
@@ -291,9 +291,31 @@ impl crate::private::Sealed for Async {}
 pub(crate) mod private {
     pub trait Sealed {}
 
+    #[non_exhaustive]
+    #[doc(hidden)]
+    /// Magical incantation to gain access to internal APIs.
     pub struct Internal;
+
+    impl Internal {
+        /// Obtain magical powers to access internal APIs.
+        ///
+        /// # Safety
+        ///
+        /// By calling this function, you accept that you are using an internal
+        /// API that is not guaranteed to be documented, stable, working
+        /// and may change at any time.
+        ///
+        /// You declare that you have tried to look for other solutions, that
+        /// you have opened a feature request or an issue to discuss the
+        /// need for this function.
+        pub unsafe fn conjure() -> Self {
+            Self
+        }
+    }
 }
 
+pub use private::Internal;
```

Comment:
I can't recall, but does this `pub use` need to be `doc(hidden)` too?

Reply:
Probaby, I'm checking/fixing